server/routes/api.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express')
+const router = express.Router()
+const axios = require('axios')
+const moment = require ("moment")
+const City = require("../model/City")
+
+const apiKey = "f2d2c49051fb06b13f554f1bea9d785e"
+const getUrl = function (cityName) {
+    return `http://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&appid=${apiKey}`
+}
+const cityObj = function (cityData) {
+    let mapedData = {
+        name: cityData.data.name.toLowerCase(),
+        temperature: cityData.data.main.temp,
+        condition: cityData.data.weather[0].description,
+        icon: cityData.data.weather[0].icon,
+        updatedAt:  moment(new Date()).format('LLLL')
+    }
+    return mapedData
+}
+    router.get('/city/:cityName', async function (req, res) {
+        const cityName = req.params.cityName
+        const lon = req.query.lon
+        const lat = req.query.lat
+        if (lon && lat) {
+            try {
+                const weatherData = await axios.get(`http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=metric&appid=${apiKey}`)
+                let mapedData = {
+                    name: weatherData.data.name.toLowerCase(),
+                    temperature: weatherData.data.main.temp,
+                    condition: weatherData.data.weather[0].description,
+                    icon: weatherData.data.weather[0].icon,
+                    lon: weatherData.data.coord.lon,
+                    lat: weatherData.data.coord.lat,
+                    updatedAt: moment(new Date()).format('LLLL')
```

Comment:
I thought you were formattting the date on the client side. If you are, you don't need to do so here as well.

Reply:
That's what i tried first.
I'm not sure why, it's not storing the data on DB that way.
I think i asked Lior about that.
On DB it saved with this format 2020-09-22T08:01:00.000Z

Reply:
Yes, Mongo stores dates like that, and since you've defined it as a date in your schema, then it is going into the Mongo format. In general, the format for the date is only important for aesthetics (how the user sees it) so it should be done while rendering. No need to save the formatted date.